test/test_pipeline_extractive_qa.py

```diff
@@ -98,3 +101,48 @@ def test_extractive_qa_answers_with_translator(
     assert (
         prediction["answers"][0].context == "My name is Carla and I live in Berlin"
     )
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize("retriever_with_docs", ["elasticsearch", "dpr", "embedding"], indirect=True)
+@pytest.mark.parametrize("document_store_with_docs", ["elasticsearch"], indirect=True)
+@pytest.mark.parametrize("reader", ["farm"], indirect=True)
+def test_extractive_qa_es_authentication(reader, retriever_with_docs, document_store_with_docs: ElasticsearchDocumentStore):
+    if isinstance(retriever_with_docs, (DensePassageRetriever, EmbeddingRetriever)):
+        document_store_with_docs.update_embeddings(retriever=retriever_with_docs)
+    mock_client = Mock(wraps=document_store_with_docs.client)
+    document_store_with_docs.client = mock_client
+    auth_headers = {'Authorization': 'Basic YWRtaW46cm9vdA=='}
+    pipeline = ExtractiveQAPipeline(reader=reader, retriever=retriever_with_docs)
+    prediction = pipeline.run(
+        query="Who lives in Berlin?", params={"Retriever": {"top_k": 10, "headers": auth_headers}, "Reader": {"top_k": 3}},
+    )
+    assert prediction is not None
+    assert len(prediction["answers"]) == 3
+    mock_client.search.assert_called_once()
+    args, kwargs = mock_client.search.call_args
+    assert "headers" in kwargs
+    assert kwargs["headers"] == auth_headers
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize("retriever_with_docs", ["tfidf"], indirect=True)
+@pytest.mark.parametrize("reader", ["farm"], indirect=True)
+def test_extractive_qa_document_store_authentication(reader, retriever_with_docs, document_store_with_docs):
```

Comment:
Why did you split this into two tests? 
In case it's not an ES docstore, we basically just test that passing headers doesn't throw an error? For this, the test seems quite expansive (loading in the whole pipeline etc.). 
In case it's an elastic store, we basically test the same thing as above (just this time with tf-idf)?

Reply:
First one ensures all retrievers compatible with ElasticsearchDocumentStore and actually searching with es propagate the headers correctly. Second is intended to ensure that all DocumentStores can deal with the headers param. In case of ElasticsearchDocumentStore TFIDF Retriever does not 'search' with ES (e.g. call ElasticSearch's client.search()) but only calls count() on the document store. So that is also an implicit test, that not only client.search() gets the headers propagated correctly.